### PR TITLE
Add unic-ucd-case component

### PR DESCRIPTION
**Pull request by behnam**
Add `unic-ucd-case` component with implementation for following
attributes, all using the new binary form for `char_property` marco:

* `Lowercase`
* `Uppercase`
* `Cased`
* `Case_Ignorable`
* `Changes_When_Lowercased`
* `Changes_When_Uppercased`
* `Changes_When_Titlecased`
* `Changes_When_Casefolded`
* `Changes_When_Casemapped`

Expose the new API (types and shorthand functions) to `unic-ucd` and add
a use-case to `basics_test` file and docs.

Tracker: <https://github.com/behnam/rust-unic/issues/153>

### CODE REVIEW

File: unic/ucd/case/src/changes_when_uppercased.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// Copyright 2015 The Servo Project Developers.
```
Comment (CAD97) [inline unic/ucd/case/src/changes_when_uppercased.rs]: Just a small question: what part of this file comes from Servo libraries? I'm supposing the tests?

**Comment by behnam** [comment]
Thanks, @CAD97!

bors: r+

**Comment by behnam** [comment]
bors: ping

**Comment by behnam** [comment]
bors: r+

**Pull request merged**